STM32F1/libraries/SPI/src/SPI.cpp

```diff
@@ -86,6 +86,14 @@ static const spi_pins board_spi_pins[] __FLASH__ = {
 #endif
 };
 
+static const spi_pins board_alt_spi_pins[] __FLASH__ = {
+#if BOARD_NR_SPI >= 1 && defined(BOARD_SPI1_ALT_MOSI_PIN)
```

Comment:
I'd tend to agree, but this isn't a pattern I've invented here -- if you have a look at this exact file, you'll find this exact check occurring in many places.  Which would be more appropriate -- following a well-trodden pattern that neither of us have a good explanation for, or my jumping to a conclusion that I know more about the problem than those who came before me?  I'd be glad to change this, but this is essentially free since it's calculated at compilation time, and it seems like somebody has been quite careful to ensure that this check occurs in similar situations elsewhere.

Reply:
Oh, I think I see what you mean -- if `BOARD_SPI1_ALT_MOSI_PIN` is defined, we're essentially declaring that that SPI port exists.  You're right; I agree!